client/blocks/author-selector/index.jsx

```diff
@@ -1,30 +1,31 @@
 /**
  * External dependencies
  */
-var ReactDom = require( 'react-dom' ),
-	React = require( 'react' ),
-	debug = require( 'debug' )( 'calypso:author-selector' ),
-	trim = require( 'lodash/trim' );
+import ReactDom from 'react-dom';
+import React from 'react';
+import debugModule from 'debug';
+import { trim } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var Popover = require( 'components/popover' ),
-	PopoverMenuItem = require( 'components/popover/menu-item' ),
-	SiteUsersFetcher = require( 'components/site-users-fetcher' ),
-	UserItem = require( 'components/user' ),
-	Gridicon = require( 'components/gridicon' ),
-	InfiniteList = require( 'components/infinite-list' ),
-	UsersActions = require( 'lib/users/actions' ),
-	Search = require( 'components/search' ),
-	hasTouch = require( 'lib/touch-detect' ).hasTouch;
+import Popover from 'components/popover';
+import PopoverMenuItem from 'components/popover/menu-item';
+import SiteUsersFetcher from 'components/site-users-fetcher';
+import UserItem from 'components/user';
+import Gridicon from 'components/gridicon';
+import InfiniteList from 'components/infinite-list';
+import UsersActions from 'lib/users/actions';
+import Search from 'components/search';
+import { hasTouch } from 'lib/touch-detect';
 
 /**
  * Module variables
  */
-var instance = 0, SwitcherShell;
+const debug = debugModule( 'calypso:author-selector' );
+let instance = 0;
 
-SwitcherShell = React.createClass( {
+const SwitcherShell = React.createClass( {
```

Comment:
Have you considered refactoring this as an ES6 class instead of using `React.createClass`?

Also, if you name this `AuthorSwitcherShell`, you can remove the `displayName` line below.